src/DotLiquid.Tests/DropTests.cs

```diff
@@ -353,6 +370,40 @@ public void TestAccessContextFromDrop()
             Assert.AreEqual("123", Template.Parse("{% for a in dummy %}{{ context.loop_pos }}{% endfor %}").Render(Hash.FromAnonymousObject(new { context = new ContextDrop(), dummy = new[] { 1, 2, 3 } })));
         }
 
+        [Test]
+        public void TestCancelBlockingOperationDrop()
+        {
+            using (var cancellationTokenSource = new CancellationTokenSource())
+            {
+                var cancellationToken = cancellationTokenSource.Token;
+                var context = new Context(new List<Hash>(),

```

Comment:
Nit, on coding style: consider also putting the first argument on its on line when using multiple lines. And consider using named parameters when 1. using multiple lines, or 2. using constant input like this 0, or 3. using multiple parameters of same type hence their order will be ambiguous to human reader without their name.